sdk/storage/Azure.Storage.Files/src/FileClient.cs

```diff
@@ -1682,6 +1682,167 @@ public virtual FileClient WithSnapshot(string shareSnapshot)
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual async Task<Response<StorageFileUploadInfo>> UploadRangeFromUrlAsync(
+            Uri sourceUri,
+            HttpRange range,
+            HttpRange sourceRange,
+            CancellationToken cancellationToken = default) =>
+            await this.UploadRangeFromUrlInternal(
+                sourceUri,
+                range,
+                sourceRange,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="UploadRangeInternal"/> operation writes a range from an Azure File to another Azure file. 
+        /// This API is supported only for version 2019-02-02 and higher.
+        /// </summary>
+        /// <param name="sourceUri">
+        /// Required. Specifies the URL of the source file, up to 2 KB in length.
+        /// If source is an Azure blob or Azure file, it must either be public or must be authenticated via a 
+        /// shared access signature. If the source is public, no authentication is required to perform the operation.
+        /// </param>
+        /// <param name="range">
+        /// Specifies the range of bytes to be written. Both the start and end of the range must be specified.
+        /// </param>
+        /// <param name="sourceRange">
+        /// Specifies the range of bytes to be written from. Both the start and end of the range must be specified.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{FileInfo}"/> describing the
+        /// state of the file.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<StorageFileUploadInfo>> UploadRangeFromUrlInternal(
+            Uri sourceUri,
+            HttpRange range,
+            HttpRange sourceRange,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (this.Pipeline.BeginLoggingScope(nameof(FileClient)))
+            {
+                this.Pipeline.LogMethodEnter(
+                    nameof(FileClient),
+                    message:
+                    $"{nameof(this.Uri)}: {this.Uri}\n" +
+                    $"{nameof(sourceUri)}: {sourceUri}");
+                try
+                {
+                    var response = await FileRestClient.File.UploadRangeFromURLAsync(
```

Comment:
Is the model typed returned by `FileRestClient.File.UploadRangeFromURLAsync` public?  If so we should hide it.

Reply:
FileRestClient.File.UploadRangeFromURLAsync() returns a FileUploadRangeFromURLResult, which is generated and public.  How do I go about hiding it?

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files/src/Generated/FileRestClient.cs#L6146

Reply:
fixed.